### PR TITLE
feat(eventing): add io-engine events

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
Added io-engine start, stop, shutdown events.
Reactor freeze, unfreeze events.
Nexus shutdown event.

### CODE REVIEW

File: apis/events/protobuf/v1/event.proto [hunk 1]
```diff
@@ -186,3 +195,17 @@ message HostInitiatorEventDetails {
...
+// Reactor event details
+message ReactorEventDetails {
+  // The logical core this reactor is created on
+  uint64 lcore = 1;
+  // Reactor state
+  string state = 2;
+}
+
+// Nexus shutdown event details
+message NexusShutdownEventDetails {
```
Comment (tiagolobocastro) [inline apis/events/protobuf/v1/event.proto]: Shouldn't it be IoEngine Shutdown? (it's the whole process being shutdown, not just the nexuses)
Then by its side we can have a per-nexus shutdown event?

Comment (datacore-vvarakantham) [reply apis/events/protobuf/v1/event.proto]: Should we have two separate events for io-engine stop and shutdown?

Comment (datacore-vvarakantham) [reply apis/events/protobuf/v1/event.proto]: Added IoEngine shutdown and per nexus shutdown events.

File: apis/events/protobuf/v1/event.proto [hunk 2]
```diff
@@ -186,3 +193,11 @@ message HostInitiatorEventDetails {
   // Target uuid
   string uuid = 4;
 }
+
+// Reactor event details
+message ReactorEventDetails {
```
Comment (tiagolobocastro) [inline apis/events/protobuf/v1/event.proto:207]: Frozen reactors can also unfreeze, probably need to take into account here so we can get events for both freeze and unfreeze?

Comment (tiagolobocastro) [reply apis/events/protobuf/v1/event.proto:207]: Or maybe just have a ReactorUnfreeze above

Comment (datacore-vvarakantham) [reply apis/events/protobuf/v1/event.proto:207]: Added ReactorUnfreeze EventAction now.

**Review by tiagolobocastro — APPROVED** [review]

**Review by sinhaashish — APPROVED** [review]

**Pull request merged**